### ISSUE TITLE
CA2245- issue not found

### ISSUE THREAD

**Issue by MarkoFilip** [dotnet/roslyn-analyzers#3499]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.8 (latest release) 

#### Diagnostic ID

[CA2245]

#### Repro steps

1. Define property as 
    public int Level
    {
      get 
      { 
        return this.Level; 
      }
      set 
      { 
        this.Level = value + 1; 
      }
    }

#### Expected behavior
Warning for issue CA2245 is displayed

#### Actual behavior
No errors/warnings are reported.

**Comment by Evangelink** [comment]
Hi @MarkoFilip, the rule [CA2245](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2245) is only looking at self-assignment while your case is more about infinite recursion which is rule CA2011](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2011).

**Comment by mavasani** [comment]
I confirmed @Evangelink's comment. Above code generates CA2011 and the indeed causes a stack overflow exception at runtime:

```csharp
    public class Class1
    {
        public int Level
        {
            get
            {
                return this.Level;
            }
            set
            {
                this.Level = value + 1;
            }
        }

        public static void Main()
        {
            // Stack overflow
            new Class1().Level = 0;
        }
    }
```

**Comment by MarkoFilip** [comment]
Confirmed on my side also. The issue was that the rule is only implemented
in new beta release, so I haven't noticed it.
Sorry, for taking your time.
Br, Marko
